File: src/types/constraint.ts
export interface ConstraintRule {
  id: number;
  ruleType: string;
  description: string;
  weight: number;
  isActive: boolean;
  parameters: string;
  sortOrder: number;
  scopeType: string;
  scopeId: number | null;
}

export interface NewConstraintRule {
  ruleType: string;
  description: string;
  weight?: number;
  isActive?: boolean;
  parameters?: string;
  scopeType?: string;
  scopeId?: number | null;
}

export interface ConstraintOrderUpdate {
  id: number;
  sortOrder: number;
}

File: src-tauri/src/models/constraint.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ConstraintRule {
    pub id: i64,
    pub rule_type: String,
    pub description: String,
    pub weight: f64,
    pub is_active: bool,
    pub parameters: String,
    pub sort_order: i32,
    pub scope_type: String,
    pub scope_id: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NewConstraintRule {
    pub rule_type: String,
    pub description: String,
    pub weight: Option<f64>,
    pub is_active: Option<bool>,
    pub parameters: Option<String>,
    pub scope_type: Option<String>,
    pub scope_id: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ConstraintOrderUpdate {
    pub id: i64,
    pub sort_order: i32,
}
